canonical/maas-ui
# add websocket debug in development

## Done

- enable websocket debug logger in development

_Note: all env vars can be overwritten using the .env.local file (e.g. if you wanted to switch this off on your machine)_

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the demo
- verify there are no websocket debug logs in the browser console
- run this PR locally
- verify you can see websocket debug logs in the browser console

## Fixes

Fixes: #4413  

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots
<img width="1189" alt="image" src="https://user-images.githubusercontent.com/7452681/191461165-ec6d5026-145b-4722-9677-6f992fbd4b92.png">
